src/spikeinterface/postprocessing/amplitude_scalings.py

```diff
@@ -526,133 +587,137 @@ def fit_collision(
-#             sparse_indices = np.union1d(sparse_indices, sparse_indices_i)
-#     else:
-#         sparse_indices = np.unique(overlapping_spikes["channel_index"])
-
-#     channel_ids = recording.channel_ids[sparse_indices]
-
-#     center_spike = overlapping_spikes[0]
-#     max_delta = np.max(
-#         [
-#             np.abs(center_spike["sample_index"] - np.min(overlapping_spikes[1:]["sample_index"])),
-#             np.abs(center_spike["sample_index"] - np.max(overlapping_spikes[1:]["sample_index"])),
-#         ]
-#     )
-#     sf = max(0, center_spike["sample_index"] - max_delta - cut_out_samples)
-#     ef = min(
-#         center_spike["sample_index"] + max_delta + cut_out_samples,
-#         recording.get_num_samples(segment_index=center_spike["segment_index"]),
-#     )
-#     tr_overlap = recording.get_traces(start_frame=sf, end_frame=ef, channel_ids=channel_ids, return_scaled=True)
-#     ts = np.arange(sf, ef) / recording.sampling_frequency * 1000
-#     max_tr = np.max(np.abs(tr_overlap))
-
-#     for ch, tr in enumerate(tr_overlap.T):
-#         _ = ax.plot(ts, tr + 1.2 * ch * max_tr, color="k")
-#         ax.text(ts[0], 1.2 * ch * max_tr - 0.3 * max_tr, f"Ch:{channel_ids[ch]}")
-
-#     used_labels = []
-#     for i, spike in enumerate(overlapping_spikes):
-#         label = f"U{spike['unit_index']}"
-#         if label in used_labels:
-#             label = None
-#         else:
-#             used_labels.append(label)
-#         ax.axvline(
-#             spike["sample_index"] / recording.sampling_frequency * 1000, color=f"C{spike['unit_index']}", label=label
-#         )
-
-#     if scalings is not None:
-#         fitted_traces = np.zeros_like(tr_overlap)
-
-#         all_templates = we.get_all_templates()
-#         for i, spike in enumerate(overlapping_spikes):
-#             template = all_templates[spike["unit_index"]]
-#             overlap_index = np.where(spikes == spike)[0][0]
-#             template_scaled = scalings[overlap_index] * template
-#             template_scaled_sparse = template_scaled[:, sparse_indices]
-#             sample_start = spike["sample_index"] - we.nbefore
-#             sample_end = sample_start + template_scaled_sparse.shape[0]
-
-#             fitted_traces[sample_start - sf : sample_end - sf] += template_scaled_sparse
-
-#             for ch, temp in enumerate(template_scaled_sparse.T):
-#                 ts_template = np.arange(sample_start, sample_end) / recording.sampling_frequency * 1000
-#                 _ = ax.plot(ts_template, temp + 1.2 * ch * max_tr, color=f"C{spike['unit_index']}", ls="--")
-
-#         for ch, tr in enumerate(fitted_traces.T):
-#             _ = ax.plot(ts, tr + 1.2 * ch * max_tr, color="gray", alpha=0.7)
-
-#         fitted_line = ax.get_lines()[-1]
-#         fitted_line.set_label("Fitted")
-
-#     ax.legend()
-#     ax.set_title(f"Spike {spike_index} - sample {center_spike['sample_index']}")
-#     return ax
+### Debugging ###
```

Comment:
Should this be commented again?

Reply:
I think this is a good time to discuss, for me it is confusing to have blocks of commented code in the main codebase. It also leads to problems as when the code is wanted to be used, it cannot be because it's gone out of date (which has happened here). For me better options are (not in order of preference)  1) full inclusion as a widget + testing 2) use as debugging tool with quick tests to ensure it doesn't become unuseable 3) factoring to a private repo / deletion

Reply:
I would keep them there uncommented. Hopefully, as we maintain the extension we can update the plotting functions as we go :)

Reply:
I would rather have deletion, I think this is a convenience for Sam workflow. Maybe something to discuss on an issue though. I was just trying to make the diff smaller to no distract the review.

Reply:
I wrote this without updating and did not see that @alejoe91 commented before me.  That is fine as well.